src/ucp/rma/flush.c

```diff
@@ -201,14 +221,32 @@ ucs_status_t ucp_ep_flush_progress_pending(uct_pending_req_t *self)
 
     ucs_assert(!(req->flags & UCP_REQUEST_FLAG_COMPLETED));
 
+    if (ep->flags & UCP_EP_FLAG_RESTART_FLUSH) {
+        uct_worker_progress_register_safe(ep->worker->uct,
+                                          ucp_ep_flush_resume_slow_path_callback,
+                                          req, 0, &req->send.flush.prog_id);
+        /* Request was detached from pending and should be scheduled again */
+        ucs_assert((req->send.lane == 0));
+        ucs_assert(!req->send.flush.sw_started);
+        req->send.lane = UCP_NULL_LANE;
+        return UCS_OK;
+    }
+
+    ucs_assert(lane != UCP_NULL_LANE);
+    ucs_assert(!(req->send.flush.started_lanes & UCS_BIT(lane)));
+
     status = uct_ep_flush(ucp_ep_get_lane(ep, lane), req->send.flush.uct_flags,
                           &req->send.state.uct_comp);
-    ucs_trace("flushing ep %p lane[%d]=%p: %s", ep, lane,
-              ucp_ep_get_lane(ep, lane), ucs_status_string(status));
+    ucs_trace_req("req %p: flush ep %p lane[%d]=%p: %s", req, ep, lane,
+                  ucp_ep_get_lane(ep, lane), ucs_status_string(status));
     if (status == UCS_OK) {
+        req->send.flush.started_lanes |= UCS_BIT(lane);
```

Comment:
ohhh...
maybe add wrapper for --comp.count + updating started lanes and reuse it in lines 115-120?

Reply:
yeah, let's see how CI goes... ep flush looks like corner circle or inf polygon :) then clean up the code